units/host_status.go

```diff
@@ -106,19 +112,44 @@ func (j *cloudHostReadyJob) Run(ctx context.Context) {
 			for i := range hosts {
 				j.AddError(errors.Wrap(setCloudHostStatus(ctx, m, hosts[i], statuses[i]), "error settings cloud host status"))
 			}
-			continue
+			continue clientsLoop
 		}
 		for _, h := range hosts {
 			hostStatus, err := m.GetInstanceStatus(ctx, &h)
 			if err != nil {
 				j.AddError(errors.Wrapf(err, "error checking instance status of host %s", h.Id))
-				continue
+				continue clientsLoop
 			}
 			j.AddError(errors.Wrap(setCloudHostStatus(ctx, m, h, hostStatus), "error settings instance statuses"))
 		}
 	}
 }
 
+func (j *cloudHostReadyJob) terminateUnknownHosts(ctx context.Context, awsErr string) error {
+	pieces := strings.Split(awsErr, "'")
+	if len(pieces) != 3 {
+		return errors.Errorf("unexpected format of AWS error: %s", awsErr)
+	}
+	instanceIDs := strings.Split(pieces[1], ",")
+	grip.Warning(message.Fields{
+		"message": "host IDs not found in AWS, will terminate",
+		"hosts":   instanceIDs,
+	})
+	catcher := grip.NewBasicCatcher()
+	for _, hostID := range instanceIDs {
```

Comment:
Just before this for loop, would it be useful to validate that the host IDs we have parsed look like AWS host IDs? We are treating AWS errors like an API, which presumably they are not. If they change something, it'd be helpful to have some validation here so that we don't burn through request limits by passing strings to the API that aren't host IDs.

Reply:
I'm not sure that doing so would be useful. The inputs to this job are the output IDs of provisioning-create-host, which gets the IDs from amazon. If they change the format, having a check here likely wouldn't do anything